### ISSUE TITLE
Unexpected division result

### ISSUE THREAD

**Issue by JanPel** [paupino/rust-decimal#326]
When using this great project we encountered the following unexpected result.  The division result seems way smaller then what one would expect. It seems to have something to do with the precision, because if we remove the first 1 from the divisor the error doesn't occur. Not sure whether this is the expected behaviour?  My guess is that we cross into the `d2.mid` side of the divisor calculation, because `11_939_280_000 > u32::MAX` and `1_939_280_000 < u32::MAX`. 
```  
    #[test]
    fn test_division() {
        use rust_decimal::Decimal;
        use std::str::FromStr;

        let divisor = Decimal::from_str("119392.80000").unwrap();
        let dividend = Decimal::from_str("122.17103945523733226517123974").unwrap();
        assert_eq!(
            dividend / divisor,
            Decimal::from_str("0.0000000000061197796992878323").unwrap()
        );
    }

```

**Comment by paupino** [comment]
Hi @JanPel,

What version of the library are you using? Using both `1.10.3` and `master` I get: 

`122.17103945523733226517123974 / 119392.80000 = 0.0010232697403464642111180175`

Test case reproduced in [this branch](https://github.com/paupino/rust-decimal/blob/63b13679b580d021ecbe86be13e4d2bb7c60a94b/tests/decimal_tests.rs#L574-L578).

Please note, there was a bug that was fixed in `1.10.2` which may have caused some of the scenarios you're seeing (https://github.com/paupino/rust-decimal/blob/master/VERSION.md#1102).

If you are getting this in the current version, can you please let me know your feature configuration too? Thanks!

**Comment by JanPel** [comment]
Yeah was still using version `1.7.0`, just tried running it with `1.10.3` and got the expected result now. Should have tried updating the library first before making an issue. Thanks a lot for the quick response!
